pangeo_forge_orchestrator/meta_types/bakery.py

```diff
@@ -0,0 +1,240 @@
+    "azure.southafricawest",
+    "azure.australiacentral",
+    "azure.australiacentral2",
+    "azure.australiasoutheast",
+    "azure.japanwest",
+    "azure.jioindiacentral",
+    "azure.koreasouth",
+    "azure.southindia",
+    "azure.westindia",
+    "azure.canadaeast",
+    "azure.francesouth",
+    "azure.germanynorth",
+    "azure.norwaywest",
+    "azure.swedensouth",
+    "azure.switzerlandwest",
+    "azure.ukwest",
+    "azure.uaecentral",
+    "azure.brazilsoutheast",
+]
+s3_default_cache_types = Literal["bytes", "none"]
+
+# turn fsspec's list of known_implementations into object which pydantic can validate against
+KnownImplementations = Literal[tuple(list(known_implementations))]  # type: ignore
+
+FARGATE_CLUSTER = "aws.fargate"
+AKS_CLUSTER = "azure.aks"
+clusters = Literal[FARGATE_CLUSTER, AKS_CLUSTER]  # type: ignore
+
+PANGEO_FORGE_BAKERY_DATABASE = (
+    "https://raw.githubusercontent.com/pangeo-forge/bakery-database/main/bakeries.yaml"
+)
+# ensure that secrets are passed as env var names enclosed in curly braces, e.g. `"{MY_AWS_KEY}"`
+env_var_name = constr(regex=r"{(.*?)}")
+# accepts only strings of integer values 0-9
+run_identifier = constr(regex=r"^\d+$")
+# see https://github.com/pangeo-forge/roadmap/pull/34; the "@" delimiter is introduced here
+feedstock_name_with_version = constr(regex=r".*-feedstock@(0|[1-9]\d*)\.(0|[1-9]\d*)$")
+# TODO: possibly move recipe name validation into `pangeo_forge_orchestrator.meta_types.meta`
+# note that the string prior to the optional ":" delimiter must be a valid python identifier, but
+# the string after ":" is a dictionary key, and therefore does not have this requirement
+recipe_or_dictobj_identifier = constr(
+    regex=r"^([A-Za-z][A-Za-z0-9_]*)([:][A-Za-z][A-Za-z0-9_-]*)?$"
+)
+
+
+class StorageOptions(BaseModel):
+    key: Optional[env_var_name] = None  # type: ignore
+    secret: Optional[env_var_name] = None  # type: ignore
+    anon: Optional[bool] = None
+    # TODO: ensure `v` in `{"client_kwargs": {"headers": {"Authorization": v}}}` is `env_var_name`?
+    client_kwargs: Optional[dict] = None
+    default_cache_type: Optional[s3_default_cache_types] = None
+    default_fill_cache: Optional[bool] = None
+    use_listings_cache: Optional[bool] = None
+    username: Optional[env_var_name] = None  # type: ignore
+    password: Optional[env_var_name] = None  # type: ignore
+
+    class Config:
+        extra = "forbid"
+
+
+@dataclass
+class Endpoint:
+    protocol: KnownImplementations  # type: ignore
+    storage_options: Optional[StorageOptions] = None
+    prefix: Optional[str] = None
+
+
+@dataclass
+class Target:
+    region: regions
+    private: Optional[Endpoint] = None
+    public: Optional[Endpoint] = None
+    description: Optional[str] = None
+    prefix: Optional[str] = None
+
+
+@dataclass
+class FargateClusterOptions:
+    vpc: str
+    cluster_arn: str
+    task_role_arn: str
+    execution_role_arn: str
+    security_groups: List[str]
```

Comment:
Why do we need to know about this stuff? It seems like a purely internal implementation detail of the bakery. cc @sharkinsspatial